[Bug]: T2/Snappi_tests: Use round() instead of ceil() or int() for approximate calculations

### Issue Description

Many of the scripts use ceil() or int() to find some approximate values. This sometimes results in unnecessary failures. One example is below:

```
        background_loss = 0
        for row in rows:
            if 'Test Flow' in row.name:
                pytest_assert(int(row.loss) == 0, "FAIL: {} must have 0% loss".format(row.name))
            elif 'Background Flow' in row.name:
                background_loss += float(row.loss)
>       pytest_assert(int(background_loss/4) == 10, "Each Background Flow must have an avg of 10% loss ")
E       Failed: Each Background Flow must have an avg of 10% loss

background_loss = 39.998999999999995
row        = <snappi.snappi.FlowMetric object at 0x7f90adb90d40>
rows       = <snappi.snappi.FlowMetricIter object at 0x7f90adce1340>
rx_port    = {'api_server_ip': '172.27.147.46', 'asic_type': 'cisco-8000', 'asic_value': 'asic0', 'card_id': '1', ...}
tx_port    = [{'api_server_ip': '172.27.147.46', 'asic_type': 'cisco-8000', 'asic_value': 'asic0', 'card_id': '1', ...}, {'api_server_ip': '172.27.147.46', 'asic_type': 'cisco-8000', 'asic_value': 'asic0', 'card_id': '1', ...}]

snappi_tests/multidut/pfc/files/m2o_fluctuating_lossless_helper.py:418: Failed
```

As we can see, the background_loss value is nearly 40, but the int() call changes it to 9 instead of 10, and above check fails. We should use round() function as below:
```
>>> background_loss = 39.998999999999995
>>> int(background_loss/4)
9
>>> round(background_loss/4)
10
>>> 
```
@selldinesh , @kamalsahu0001 , @sdszhang , @auspham : FYI

### Results you see

Test fails on approximate calculations.

### Results you expected to see

The test should pass.

### Is it platform specific

generic

### Relevant log output

```shell
Pls see above.
There are others as well:

sonic@sonic-ucs-m4-3:~/rraghav/Ixia/whitebox/vanilla_202405_nov_21_2024/sonic-mgmt/tests/snappi_tests/multidut$ grep -rl ceil . | grep -v pyc
./bgp/files/bgp_outbound_helper.py
./pfc/files/m2o_fluctuating_lossless_helper.py
./pfc/files/m2o_oversubscribe_lossless_lossy_helper.py
./pfc/files/lossless_response_to_throttling_pause_storms_helper.py
./pfc/files/m2o_oversubscribe_lossless_helper.py
./pfc/files/m2o_oversubscribe_lossy_helper.py
./pfc/files/lossless_response_to_external_pause_storms_helper.py
./pfcwd/files/pfcwd_multidut_basic_helper.py
./pfcwd/files/pfcwd_multidut_burst_storm_helper.py
./pfcwd/files/pfcwd_multidut_multi_node_helper.py
sonic@sonic-ucs-m4-3:~/rraghav/Ixia/whitebox/vanilla_202405_nov_21_2024/sonic-mgmt/tests/snappi_tests/multidut$ 
```
```


### Output of `show version`

_No response_

### Attach files (if any)

_No response_

--- a/tests/snappi_tests/multidut/pfc/files/lossless_response_to_external_pause_storms_helper.py
+++ b/tests/snappi_tests/multidut/pfc/files/lossless_response_to_external_pause_storms_helper.py
@@ -154,7 +154,7 @@ def run_lossless_response_to_external_pause_storms_test(api,
     total_rx_pkts = rx_pkts_1 + rx_pkts_2
     # Calculate the drop percentage
     drop_percentage = 100 * pkt_drop / total_rx_pkts
-    pytest_assert(ceil(drop_percentage) == 0, 'FAIL: There should be no packet drops in ingress dut counters')
+    pytest_assert(round(drop_percentage) == 0, 'FAIL: There should be no packet drops in ingress dut counters')
 
     verify_external_pause_storm_result(flow_stats,
                                        tx_port,

--- a/tests/snappi_tests/multidut/pfc/files/lossless_response_to_throttling_pause_storms_helper.py
+++ b/tests/snappi_tests/multidut/pfc/files/lossless_response_to_throttling_pause_storms_helper.py
@@ -161,7 +161,7 @@ def run_lossless_response_to_throttling_pause_storms_test(api,
     total_rx_pkts = rx_pkts_1 + rx_pkts_2
     # Calculate the drop percentage
     drop_percentage = 100 * pkt_drop / total_rx_pkts
-    pytest_assert(ceil(drop_percentage) == 0, 'FAIL: There should be no packet drops in ingress dut counters')
+    pytest_assert(round(drop_percentage) == 0, 'FAIL: There should be no packet drops in ingress dut counters')
 
     """ Verify Results """
     verify_throttling_pause_storm_result(flow_stats,

--- a/tests/snappi_tests/multidut/pfc/files/m2o_fluctuating_lossless_helper.py
+++ b/tests/snappi_tests/multidut/pfc/files/m2o_fluctuating_lossless_helper.py
@@ -1,5 +1,4 @@
 import logging                                                                          # noqa: F401
-from math import ceil
 from tests.common.helpers.assertions import pytest_assert, pytest_require               # noqa: F401
 from tests.common.fixtures.conn_graph_facts import conn_graph_facts, fanout_graph_facts  # noqa: F401
 from tests.common.snappi_tests.snappi_helpers import get_dut_port_id                     # noqa: F401
@@ -143,7 +142,7 @@ def run_m2o_fluctuating_lossless_test(api,
     total_rx_pkts = rx_pkts_1 + rx_pkts_2
     # Calculate the drop percentage
     drop_percentage = 100 * pkt_drop / total_rx_pkts
-    pytest_assert(ceil(drop_percentage) == 8, 'FAIL: Drop packets must be around 8 percent')
+    pytest_assert(round(drop_percentage) == 8, 'FAIL: Drop packets must be around 8 percent')
 
     """ Verify Results """
     verify_m2o_fluctuating_lossless_result(flow_stats,

--- a/tests/snappi_tests/multidut/pfc/files/m2o_oversubscribe_lossless_helper.py
+++ b/tests/snappi_tests/multidut/pfc/files/m2o_oversubscribe_lossless_helper.py
@@ -146,7 +146,7 @@ def run_m2o_oversubscribe_lossless_test(api,
     total_rx_pkts = rx_pkts_1 + rx_pkts_2
     # Calculate the drop percentage
     drop_percentage = 100 * pkt_drop / total_rx_pkts
-    pytest_assert(ceil(drop_percentage) == 0, 'FAIL: There should be no packet drops in ingress dut counters')
+    pytest_assert(round(drop_percentage) == 0, 'FAIL: There should be no packet drops in ingress dut counters')
 
     """ Verify Results """
     verify_m2o_oversubscribe_lossless_result(flow_stats,

--- a/tests/snappi_tests/multidut/pfc/files/m2o_oversubscribe_lossless_lossy_helper.py
+++ b/tests/snappi_tests/multidut/pfc/files/m2o_oversubscribe_lossless_lossy_helper.py
@@ -151,7 +151,7 @@ def run_pfc_m2o_oversubscribe_lossless_lossy_test(api,
     total_rx_pkts = rx_pkts_1 + rx_pkts_2
     # Calculate the drop percentage
     drop_percentage = 100 * pkt_drop / total_rx_pkts
-    pytest_assert(ceil(drop_percentage) == 5, 'FAIL: Drop packets must be around 5 percent')
+    pytest_assert(round(drop_percentage) == 5, 'FAIL: Drop packets must be around 5 percent')
 
     """ Verify Results """
     verify_m2o_oversubscribe_lossless_lossy_result(flow_stats,

--- a/tests/snappi_tests/multidut/pfc/files/m2o_oversubscribe_lossy_helper.py
+++ b/tests/snappi_tests/multidut/pfc/files/m2o_oversubscribe_lossy_helper.py
@@ -151,7 +151,7 @@ def run_pfc_m2o_oversubscribe_lossy_test(api,
     total_rx_pkts = rx_pkts_1 + rx_pkts_2
     # Calculate the drop percentage
     drop_percentage = 100 * pkt_drop / total_rx_pkts
-    pytest_assert(ceil(drop_percentage) == 10, 'FAIL: Drop packets must be around 10 percent')
+    pytest_assert(round(drop_percentage) == 10, 'FAIL: Drop packets must be around 10 percent')
 
     """ Verify Results """
     verify_m2o_oversubscribe_lossy_result(flow_stats,

